### ISSUE TITLE
CHW/ANC - Create an Observation detail to collect vital signs information for a non-ANC client 

### ISSUE THREAD

**Issue by f-odhiambo** [opensrp/fhircore#528]
For non a non ANC client . Collect vitals and add them to their profile page as described in the mock.

On patient create , there will be no details on the profile page as decruibed in the mock 
<img width="432" alt="Blank Care plan" src="https://user-images.githubusercontent.com/4540684/132647559-06650520-69df-469e-bfb2-681006fc8cc8.png">

On the menu item in the profile page, click on the add vitals page that will load for you a Questionnaire to fill in 

<img width="330" alt="Menu Item" src="https://user-images.githubusercontent.com/4540684/132678513-81c3f754-7fdf-4445-94a6-ceb4f4bce47c.png">

Once the details are filled in the profile page will display 
1. Allergies
2. Health conditions and Vitals (Will need to figure out how to store nested conditions  as separate condiction resources)
3. Add vitals for now check Weight, Height, Blood Oxygen levels (Sp02 level), Blood glucose, Pulse rate, Blood pressure

<img width="328" alt="Vitals" src="https://user-images.githubusercontent.com/4540684/132678464-0d9259ca-267a-4e29-bd13-eaeab865171e.png">

**SubTasks**
1. Design the profile page 
2. Create a questionnaire that administers a survey on 
> 1. Allergies
> 2. Health conditions and Vitals (Will need to figure out how to store nested conditions  as separate condiction resources)
> 3. Add vitals for now check Weight, Height, Blood Oxygen levels (Sp02 level), Blood glucose, Pulse rate, Blood pressure
3. On click add vitals show page to collect above information 
4. On submit create 
1. Encounter to collect vital signs from the patient with a tag to the administering practitioner
2. A single or a separate observation resource for the vital signs   

**Acceptance criteria** 
1. On click new patient no vital signs present 
2. On save patient, refresh profile to have vital signs 
3. Check HAPI sync the vitals signs observation details

**Comment by RaaziaTarique** [comment]
Should we record all vital signs at same time or one at a time? @f-odhiamb
![Screenshot_1632209288](https://user-images.githubusercontent.com/89384079/134129974-bfb3ebe2-2b29-43af-aec4-12aef18a3711.png)
o

**Comment by f-odhiambo** [comment]
I think this could capture multiple at a time. For the purposes of this POC we can do one at a time

**Comment by pld** [comment]
do we have validation in the form to restrict the range for these? e.g. weight can't be negative

**Comment by RaaziaTarique** [comment]
@pld yes we can set minimum and maximum value for integer input fields.

**Comment by pld** [comment]
Cool, do you have the limits we want these fields defined? Or do we need to define those?

CC @isabelcshaw

**Comment by RaaziaTarique** [comment]
@pld I don't have such limits for any of the vital signs.

**Comment by pld** [comment]
Cool, @isabelcshaw can you note the limits we should enforce for these fields?

**Comment by isabelcshaw** [comment]
@pld @RaaziaTarique

It looks like from the other issue that this is done apart from the constraints, is that correct? I think we may not want to overdue the constraints at this point, so provided some basic guardrails below. 

Weight - integer; for now, I would just ensure that that it is a two or three digit number 
Height - integer; no more than 300 
Blood Oxygen Level - percent; cannot have an input over 100; normal pulse oximeter readings usually range from 95 to 100 percent, under 60 is considered dangerous - for the moment, however, let's leave off a bottom constraint 
Blood Pressure - two part integer; both integers no more than 200
Pulse Rate - integer no more than 300
Blood Glucose - integer no more than 1000

**Comment by RaaziaTarique** [comment]
@isabelcshaw Adding these limits.
CC: @pld @f-odhiambo

**Comment by f-odhiambo** [comment]
Added a related SDK ticket to handle min/max 

CC @RaaziaTarique @syedowaisali90 @pld @isabelcshaw
